### ISSUE TITLE
W8003 False positive

### ISSUE THREAD

**Issue by gabelton** [aws-cloudformation/cfn-lint#3130]
### CloudFormation Lint Version

0.86.0

### What operating system are you using?

mac

### Describe the bug

lint raises a W8003 for the below template, but I don't think it's true that it will always return True or False. It checks whether the value obtained from the mapping for the specified environment (!Ref Env) matches the string "No retention". If it does, then the condition evaluates to false, and if it doesn't, it evaluates to true.

### Expected behavior

W8003 isn't raised

### Reproduction template

```
Mappings:
  dwS3BucketEnvironmentConfigMap:
    # Create an entry for each environment

    prod:
      BucketName: 'digital-workspace-v2-prod'
      DeletionPolicy: Delete
    hotfix:
      BucketName: 'digital-workspace-v2-hotfix'
      DeletionPolicy: Delete
    staging:
      BucketName: 'digital-workspace-v2-staging'
      DeletionPolicy: Delete
    training:
      BucketName: 'digital-workspace-v2-training'
      DeletionPolicy: Delete
    dev:
      BucketName: 'digital-workspace-v2-dev'
      DeletionPolicy: Delete

Conditions:
  dwS3BucketApplyRetentionPolicy: !Not
    - !Equals
      - !FindInMap
        - dwS3BucketEnvironmentConfigMap
        - !Ref Env
        - RetentionMode
        - DefaultValue: No retention
      - No retention
  dwS3BucketRetentionPolicyDurationIsInDays: !Equals
    - !FindInMap
      - dwS3BucketEnvironmentConfigMap
      - !Ref Env
      - RetentionDurationType
      - DefaultValue: Days
    - Days
```

**Comment by kddejong** [comment]
Fix is in.  Closing and will be released today or tomorrow.
